elixir/apps/domain/test/domain/auth/adapters/jumpcloud/jobs/sync_directory_test.exs

```diff
@@ -0,0 +1,568 @@
+
+      # Broadcasts allow_access for it
+      policy_id = policy.id
+      group_id = group.id
+      resource_id = policy.resource_id
+      assert_receive {:allow_access, ^policy_id, ^group_id, ^resource_id}
+
+      # Deletes membership that is not found on IdP end
+      actor_id = deleted_membership.actor_id
+      group_id = deleted_membership.group_id
+      assert_receive {:delete_membership, ^actor_id, ^group_id}
+
+      # Signs out users which identity has been deleted
+      topic = "sessions:#{deleted_identity_token.id}"
+      assert_receive %Phoenix.Socket.Broadcast{topic: ^topic, event: "disconnect", payload: nil}
+
+      # Deleted group deletes all policies and broadcasts reject access events for them
+      policy_id = deleted_policy.id
+      group_id = deleted_group.id
+      resource_id = deleted_policy.resource_id
+      assert_receive {:reject_access, ^policy_id, ^group_id, ^resource_id}
+
+      # Deleted policies expire all flows authorized by them
+      flow_id = deleted_group_flow.id
+      assert_receive {:expire_flow, ^flow_id, _client_id, ^resource_id}
+
+      # Expires flows for signed out user
+      flow_id = deleted_identity_flow.id
+      assert_receive {:expire_flow, ^flow_id, _client_id, _resource_id}
+
+      # Should not do anything else
+      refute_receive {:create_membership, _actor_id, _group_id}
+      refute_received {:remove_membership, _actor_id, _group_id}
+      refute_received {:allow_access, _policy_id, _group_id, _resource_id}
+      refute_received {:reject_access, _policy_id, _group_id, _resource_id}
+      refute_received {:expire_flow, _flow_id, _client_id, _resource_id}
+    end
+
+    test "stops the sync retires on 401 error from WorkOS", %{provider: provider} do
+      bypass = Bypass.open()
+      WorkOSDirectory.override_base_url("http://localhost:#{bypass.port}")
+
+      error_message = "Error connecting to WorkOS"
+
+      response = %{"message" => "Unauthorized"}
+
+      for path <- [
+            "/directories",
+            "/directory_users",
+            "/directory_groups"
+          ] do
+        Bypass.stub(bypass, "GET", path, fn conn ->
+          conn
+          |> Plug.Conn.prepend_resp_headers([{"content-type", "application/json"}])
+          |> Plug.Conn.send_resp(401, Jason.encode!(response))
+        end)
+      end
+
+      {:ok, pid} = Task.Supervisor.start_link()
+      assert execute(%{task_supervisor: pid}) == :ok
+
+      assert updated_provider = Repo.get(Domain.Auth.Provider, provider.id)
+      refute updated_provider.last_synced_at
+      assert updated_provider.last_syncs_failed == 1
+      assert updated_provider.last_sync_error == error_message
+    end
+
+    test "persists the sync error on the provider", %{provider: provider} do
+      bypass = Bypass.open()
+
+      WorkOSDirectory.override_base_url("http://localhost:#{bypass.port}")
+      # WorkOSDirectory.mock_list_directories_endpoint(bypass)
```

Comment:
I think this comments can be removed

Reply:
Whoops!  Thanks for catching that.  I knew I was going to forget some comment.